src/main/kotlin/org/malachite/estella/task/api/TaskController.kt

```diff
@@ -19,30 +20,41 @@ import org.springframework.web.bind.annotation.*
 @RequestMapping("/api/tasks")
 class TaskController(
         @Autowired private val taskService: TaskService,
+        @Autowired private val taskStageService: TaskStageService,
         @Autowired private val organizationService: OrganizationService,
 ) {
 
-    @Deprecated("Not tested yet - draft implementation, should be tested as part of ES-162")
     @CrossOrigin
     @Transactional
     @GetMapping
     fun getTasks(
-            @RequestParam("owner", required = false) organizationUuid: String?,
-            @RequestParam("taskStage", required = false) taskStageUuid: String?,
+            @RequestParam("owner") organizationUuid: String?,
+            @RequestParam("taskStage") taskStageUuid: String?,
+            @RequestParam("devMail") devMail: String?,
             @RequestHeader(EStellaHeaders.devPassword) password: String
     ): ResponseEntity<Any> {
-        if ((organizationUuid == null && taskStageUuid == null) || (organizationUuid != null && taskStageUuid != null))
-            return ResponseEntity.badRequest().body(Message("Exactly one of parameters: organizationUuid and taskStageUuid is required"))
+        if (!areParamsValid(organizationUuid, taskStageUuid, devMail))
+            return ResponseEntity.badRequest().body(Message("Exactly one of parameters: organizationUuid, taskStageUuid and devMail is required"))
         val tasks: List<TaskDto> = organizationUuid
                 ?.let {
-                    taskService.checkDevPassword(it, password)
-                            .getTasksByOrganizationUuid(it)
+                    devMail?.let { taskStageService.getTasksByDev(it, password) }
+                            ?:let{
+                                taskStageService.checkDevPassword(organizationUuid, password)
+                                        .getTasksByOrganizationUuid(organizationUuid)
+                            }
                 }
                 ?: taskStageUuid
-                        ?.let { taskService.getTasksByTasksStage(it, password) }!!
+                        ?.let { taskStageService.getTasksByTasksStage(it, password) }!!
        return ResponseEntity.ok(tasks)
     }
 
+    private fun areParamsValid(organizationId: String?, taskStageUuid: String?, devMail: String?): Boolean {
+        if (listOfNotNull(organizationId, taskStageUuid, devMail).isEmpty()) return false
+        if (organizationId != null && taskStageUuid != null) return false
+        if (devMail != null && organizationId == null) return false
+        return true
+    }
```

Comment:
Można tu spróbować wyjść z `if (cond) true else false`, zapakować w jakąś listę booli i zrobić `.all { it }` na przykład